packages/trace/src/teleReceiver.ts

```diff
@@ -0,0 +1,435 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { FullConfig, FullResult, Location, Reporter, TestError, TestResult, TestStatus, TestStep } from '@playwright/test/types/testReporter';
+import type { Annotation, FullProject, Metadata } from '@playwright/test/src/common/types';
+import type * as reporterTypes from '@playwright/test/types/testReporter';
+
+export type JsonLocation = Location;
+export type JsonError = string;
+export type JsonStackFrame = { file: string, line: number, column: number };
+
+export type JsonConfig = {
+  rootDir: string;
+  configFile: string | undefined;
+};
+
+export type JsonProject = {
+  id: string;
+  grep: string[];
+  grepInvert: string[];
+  metadata: Metadata;
+  name: string;
+  dependencies: string[];
+  snapshotDir: string;
+  outputDir: string;
+  repeatEach: number;
+  retries: number;
+  suites: JsonSuite[];
+  testDir: string;
+  testIgnore: string[];
+  testMatch: string[];
+  timeout: number;
+};
+
+export type JsonSuite = {
+  type: 'root' | 'project' | 'file' | 'describe';
+  fileId: string;
```

Comment:
I don't see this being meaningfully used anywhere. Given that it's not in our public API, let's omit it?

Reply:
It will be used!